### PR TITLE
[GUI_sdlRender] fix HiDPI, fix preference saving

### PR DESCRIPTION
**Pull request by szlldm**

### CODE REVIEW

File: avidemux/common/ADM_render/GUI_sdlRender.cpp [hunk 1]
```diff
@@ -51,7 +51,9 @@ static std::vector<sdlDriverInfo>listOfSDLDrivers;
 class sdlRenderImpl: public VideoRenderBase
 {
   protected:
+              GUI_WindowInfo info;
```
Comment (eumagga0x2a) [inline avidemux/common/ADM_render/GUI_sdlRender.cpp:54]: I guess we don't need the full struct, just a double for scalingFactor, do we?

File: avidemux/common/ADM_render/GUI_sdlRender.cpp [hunk 2]
```diff
@@ -288,7 +299,10 @@ bool sdlRenderImpl::init( GUI_WindowInfo *window, uint32_t w, uint32_t h, float
 
     int bpp;
     int flags;
+    info=*window;
     baseInit(w,h,zoom);
+    displayWidth*=info.scalingFactor;
+    displayHeight*=info.scalingFactor;
```
Comment (eumagga0x2a) [inline avidemux/common/ADM_render/GUI_sdlRender.cpp:305]: Maybe factorized like in libvaRender?

**Comment by eumagga0x2a** [comment]
I can address the nits post-merge, if you agree.

File: avidemux/common/ADM_render/GUI_sdlRender.cpp (hunk 1 above)
Comment (szlldm) [reply avidemux/common/ADM_render/GUI_sdlRender.cpp:54]: maybe in the future other members will be needed. there is literally no gain if we only store the scalingFactor

**Pull request merged**